src/App.tsx

```diff
@@ -8,53 +9,112 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useState } from 'react';
+import { Post } from './types/Post';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [userSelectedId, setUserSelectedId] = useState<number | null>(null);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const [postsByUser, setPostsByUser] = useState<Post[]>([]);
+  const [postSelected, setPostSelected] = useState<Post | null>(null);
 
-              <Loader />
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoadind] = useState(false);
```

Comment:
Typo in state variable name: `setIsLoadind` should be `setIsLoading`. This typo is repeated in other parts of the code as well.